docs/http-api-integration.md

```diff
@@ -0,0 +1,106 @@
+> "REST provides a set of architectural constraints that, when applied as a whole, emphasizes scalability of component interactions, the generality of interfaces, independent deployment of components, and intermediary components to reduce interaction latency, enforce security, and encapsulate legacy systems"
+
+However, there is no _constraint_  on how to limit or include additional fields for `resource` response. 
+For example, techniques "Sparse Fields", "Compound Documents" and "Pagination" are not defined in the **SAC**. 
+This leaves it up to the implementers to decide how, or even if, to implement these features.
+
+This is where [JSON:API](https://jsonapi.org/format/) and [GraphQL](https://graphql.org/learn/) (among a few others) 
+provide the needed additional constraints that address the previously listed "techniques" that RESTful API does not address. 
+There are many features that JSON API and GraphQl are capable of, but we are only focusing on the ones that achieve the goal of reducing the size and number of API requests that go between the client and server.
+
+## Adding Constraints through Specifications
+
+### JSON:API
+**JSON:API** has features such as **Sparse Fields**, **Compound Documents**, **Pagination**, and **Caching**.
+
+#### Sparse Fieldset
+This is the bread and butter that allows the request to specify the fields needed.
+There are more examples on Sparse Fieldsets at [jsonapi.org](https://jsonapi.org/examples/#sparse-fieldsets)
+```
+GET /books?fields[books]=author,themes 
+```
+
+#### Compound Documents
+This feature allows for including related resources with the initial request
+```
+GET /books?fields[books]=author,themes,&include=chapter1/title 
+```
+
+#### Caching
+JSON:API takes advantage of the caching ability that HTTP has built in. This reduces the need to return data that has not changed, saving bandwidth.
+
+#### Pagination
+JSON:API handles Pagination with a standard that `links` such as: `first`, `last`, `next`, and `previous` will return subnets of sort information.
+https://jsonapi.org/format/#fetching-pagination
+
+---
+
+### GraphQL
+**GraphQL** can also reduce the fields returned by specifying them in the GraphQL Query. Keep in mind, as its name implies GraphQL is predominantly a Query Language
+
+#### Sparse Fieldsets
+```
+GET /graphql
+books(title: Hobbit) {
+  author
+  themes
+}
+
+response
+"books": {
+  author: J.R.R. Tolkien
+  themes: fantasy, fiction
+}
+```
+
+#### Caching
+GraphQL utilizes HTTP Caching and identifies when objects with globally uniquie identifiers are called again.
+https://graphql.org/learn/caching/#gatsby-focus-wrapper
+
+#### Pagination
+GraphQL suggests various way to paginate returning data by requesting a set such as `first:10 offset:20` which is the equivalent to saying "second page of 10 items". 
+Another example is `first:5 offset:$itemId`, equivalent of saying "first 5 items after the ID of this item".
+https://graphql.org/learn/pagination/
+
```

Comment:
Add a conclusion section (or suggestion)
> If you are starting out learning HTTP APIs, it may be easier to understand JSON:API. The ease of use with JSON:API is the reason why Hydrogen supports this as on option to GraphQL.